helper/threads.c

```diff
@@ -243,19 +240,15 @@ static int wait_pthread(odph_thread_t *thread)
 		return -1;
 	}
 
-	if (thread_ret) {
+	if (thread_ret)
 		ODPH_ERR("Bad exit status cpu #%i %p\n",
 			 thread->cpu, thread_ret);
-		return -1;
```

Comment:
On a glance there seems to be couple places where failures could go unnoticed in the CI after this change:
```
example/l2fwd_simple/odp_l2fwd_simple.c
helper/test/cli.c
helper/test/odpthreads.c
test/common/odp_cunit_common.c
test/performance/odp_l2fwd.c
test/performance/odp_random.c
test/validation/api/time/time.c
```

Reply:
Yes, didn't touch them until it's decided if the new proposed behavior is OK.

Reply:
I guess we could just swap to use the new version proposed in #2050 in these sort of places that don't have other means to resolve worker thread errors than the exit status.

Reply:
As discussed offline, closing this for now and leaving the current behavior as is. Potentially continued in a future release if necessary. New and improved join function in #2050.